dds/DCPS/DCPS_Utils.cpp

```diff
@@ -355,6 +383,34 @@ bool should_check_association_upon_change(const DDS::DomainParticipantQos & /*qo
   return false;
 }
 
-}}
+bool repr_to_encoding_kind(DDS::DataRepresentationId_t repr, Encoding::Kind& kind)
+{
+  switch(repr) {
+  case UNALIGNED_CDR_DATA_REPRESENTATION:
+    kind = Encoding::KIND_UNALIGNED_CDR;
+    break;
+  case DDS::XCDR_DATA_REPRESENTATION:
+    kind = Encoding::KIND_XCDR1;
+    break;
+  case DDS::XCDR2_DATA_REPRESENTATION:
+    kind = Encoding::KIND_XCDR2;
+    break;
+  default:
+    return true;
+  }
+  return false;
```

Comment:
So we return `false` for UNALIGNED_CDR, XCDR, and XCDR2, and return `true` for other (probably invalid or unsupported) representation kinds. It looks counterintuitive to me.

Reply:
Which specific part do you think is counter intuitive?

Reply:
It's a "convert a to b" function with "a" and "b" as parameters (b is by-reference of course). Shouldn't the return value be true if it succeeded and false if it failed?

Reply:
Actually I personally prefer this way, because I think it's easy for me to think of it returning if an error happened, but you're both right that is the convention and I normally do follow that here. It just looks like I mixed them up, so feel free to change it @sonndinh.

Reply:
Yes, I can change it.